api/src/main/java/io/grpc/Status.java

```diff
@@ -657,7 +657,10 @@ private static String parseAsciiStringSlow(byte[] value) {
    */
   @Override
   public boolean equals(Object obj) {
-    return super.equals(obj);
+    if (obj instanceof Status) {
```

Comment:
@amin-mlm have you checked the method javadoc? It explains the current behaviour.